caravel/models.py

```diff
@@ -644,6 +645,10 @@ def query(  # sqla
                 "Datetime column not provided as part table configuration "
                 "and is required by this type of chart"))
 
+        # Re-compile all the macros before rendering any SQL expression,
+        # so that we're sure everything is up-to-date
+        recompile_sql_macros()
```

Comment:
I don't think it's necessary to recompile all macros all the time, we only need to do this when the web server starts or the macros are altered.